### DISCUSSION TITLE
start a vanilla javascript project with vite and tailwindcss 4 beta

### DISCUSSION THREAD

**Discussion by pishev** [tailwindlabs/tailwindcss#15128] [Help]
How do you start a vanilla javascript project with vite and tailwindcss 4 beta?

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/v4-beta#installing-with-vite):

> If you're using Vite or a Vite-powered framework like SvelteKit or Remix, install Tailwind along with our new dedicated Vite plugin:
> 
> ```bash
> $ npm install tailwindcss@next @tailwindcss/vite@next
> ```
> 
> Next, add our Vite plugin to your `vite.config.ts` file:
> 
> ```ts
> import { defineConfig } from 'vite';
> import tailwindcss from '@tailwindcss/vite';
> 
> export default defineConfig({
>   plugins: [
>     tailwindcss()
>   ],
> });
> ```
> 
> Finally, import Tailwind into your main CSS file:
> 
> ```css  {{ filename: 'src/index.css' }}
> @import "tailwindcss";
> ```

**Reply by pishev** [reply]
I do not want to install any frameworks. Wasn't it clear? Vanilla JavaScript.
I can read what has been published.
Installing tailwindcss 4.0 beta with (p)npm is what I am after. Nothing else.

**Reply by adamwathan** [reply]
You asked how to install it with Vite dude:

> How do you start a vanilla javascript project **with vite** and tailwindcss 4 beta?

If you are using Vite then follow the instructions for using Vite. If you're using no bundler at all then use our CLI.
